sdc/datatypes/hpat_pandas_dataframe_rolling_functions.py

```diff
@@ -78,43 +82,175 @@
 """
 
 
-def df_rolling_method_codegen(method_name, self, args=None, kws=None):
+def params2list(params):
+    """Converts parameters from dict to list"""
+    return ['{}={}'.format(k, v) for k, v in params.items()]
+
+
+def df_rolling_params_codegen():
+    """Generate rolling parameters"""
+    params = ['window', 'min_periods', 'center', 'win_type', 'on', 'axis', 'closed']
+    return ', '.join(f'self._{p}' for p in params)
+
+
+def df_rolling_method_with_other_df_codegen(method_name, self, other, args=None, kws=None):
     args = args or []
     kwargs = kws or {}
 
-    rolling_params = ['window', 'min_periods', 'center',
-                      'win_type', 'on', 'axis', 'closed']
-    rolling_params_as_str = ', '.join(f'self._{p}' for p in rolling_params)
-    method_params = args + ['{}={}'.format(k, k) for k in kwargs]
-    method_params_as_str = ', '.join(method_params)
-
-    impl_params = ['self'] + args + ['{}={}'.format(k, v) for k, v in kwargs.items()]
+    rolling_params = df_rolling_params_codegen()
+    method_kws = {k: k for k in kwargs}
+    impl_params = ['self'] + args + params2list(kwargs)
     impl_params_as_str = ', '.join(impl_params)
 
+    data_columns = {col: idx for idx, col in enumerate(self.data.columns)}
+    other_columns = {col: idx for idx, col in enumerate(other.columns)}
+    common_columns = [col for col in data_columns if col in other_columns]
+    all_columns = [col for col in data_columns]
```

Comment:
Does the order matter? Can we just use intersection and union of sets, i.e.
```suggestion
    self_columns_set, other_columns_set = set(data_columns), set(other_columns)
    common_columns = self_columns_set.intersection(other_columns_set)
    all_columns = self_columns_set.union(other_columns_set)
```

Reply:
Nice remark, but unit tests showed that the columns order matters.